mlflow/server/js/src/model-registry/components/ModelVersionView.js

```diff
@@ -316,8 +317,18 @@ export class ModelVersionViewImpl extends React.Component {
         </a>
       );
     } else if (runInfo) {
+      if (!modelSource) {
+        throw new Error('model source is missing.');
+      }
+      const artifactPath = modelSource.match(/.+\/artifacts\/(.+)$/)[1];
```

Comment:
Does this work fine with a model logged with `mlflow.sklearn.log_model(sk_model, "artifacts/foo/bar")`?

Reply:
this is a good case for a unit test - is it possible to add a few jest tests here?

Reply:
Good catch. My first version code works fine with the case but using regex does not work with the case.
Any way to force the regex to match from left side ?

Reply:
@WeichenXu123 apologies for introducing regression, still I believe regex might be more robust compared to `indexOf`

> My first version code works fine with the case but using regex does not work with the case.

Please note that `indexOf()` returns -1 when a substring is not found, so even if `modelSource` does not contain `/artifacts/` part at all, `modelSource.indexOf('/artifacts/') + 11` will always yield 10 regardless of the input. Is it ok?

> Any way to force the regex to match from left side ?

If you want to match the beginning of the string, you can use `^` wildcard. Still, I'm willing to help with this - can you explain what are the possible/acceptable inputs for `modelSource`?

Not sure what are the acceptable set of input values

Reply:
I think valid source string might be：
`file:///path/to/mlruns/{experiment_id}/{run_id}/artifacts/{artifact_path}`
or
`mlflow-artifacts:/{experiment_id}/{run_id}/artifacts/{artifact_path}`

Reply:
I think I found solution for regex, we can use lazy mode matching:

`modelSource.match(/.+?\/artifacts\/(.+)$/)[1];` (appended `+` after the first `.+`)

Ref: https://javascript.info/regexp-greedy-and-lazy#lazy-mode

Reply:
@WeichenXu123 using lazy mode might cause `/artifacts/` being in the configurable part to be processed instead of ignored.
Please refer to
https://regex101.com/r/W08owe/1
this is the result

<img width="793" alt="Screen Shot 2023-04-04 at 16 55 19" src="https://user-images.githubusercontent.com/104438646/229832839-cd2418b1-0ff4-4478-9d5f-78f95f500b4f.png">

is it sufficient for the use case?

Reply:
@hubertzub-db 

So we cannot use lazy mode, we should match the run ID in regex, like `modelSource.match(/{run_id}\/artifacts\/(.+)$/)[1];`

Reply:
I'm a little lost now 🤔
doesn't the [example above ](https://regex101.com/r/c8aa8o/1) (one not using lazy mode) solve all use cases already? if not, let me know about any exception so we can fix it

Reply:
@hubertzub-db 

`.+\/artifacts\/(.+)` does not work with such special case:

`mlflow-artifacts:/{experiment_id}/{run_id}/artifacts/artifacts/aa/bb`

the correct artifact path for the example is `artifacts/aa/bb` but `modelSource.match(/.+\/artifacts\/(.+)$/)[1]` returns `aa/bb`

Reply:
@WeichenXu123 gotcha, didn't account for this edge case 😮 
Then I think you're right, let's use `run_id` - you can create a regexp on the fly:
```
const path = modelSource?.match(
  new RegExp(`/${runInfo.getRunUuid()}/artifacts/(.+)`),
)?.[1];
```
this should work and catch everything after first `/artifacts/` occurence following the run id, including other `artifacts` subpath.

Another tip: instead of routing the variable through `mapStateToProps()`, you can use it directly in the `resolveRunLink()` function using `modelVersion.source` since `modelVersion` is already extracted from the props.

Reply:
@hubertzub-db Thank you. I updated my PR.

I want to make `modelSource?.match(new RegExp(`/${runInfo.getRunUuid()}/artifacts/(.+)`),)?.[1]` to be a utility function `extract_artifact_path_from_model_source` and then add a unit test for it. Which place should I put this utility function ?

Reply:
eh so the problem is that our general `Utils.js` is way too big already - I propose to either create `src/model-registry/utils/VersionUtils.js` file and a corresponding test file.

Reply:
Test was added !:)